DoctorClient/DoctorClient.cpp

```diff
@@ -124,7 +124,85 @@ std::string DoctorClient::deleteAppointment(const std::string& code) {
         }
         return response.body;
     } else {
-        std::cerr << "Failed to delete appoitment " << code << ". HTTP code: " << response.code << "\n";
+        std::cerr << "Failed to delete appointment " << code << ". HTTP code: " << response.code << "\n";
+        return "Appointment Not Found";
+    }
+}
+
+std::string DoctorClient::updateAppointmentTitle(const std::string& code, const std::string& title) {
+    std::string encodedCode = encodeURIComponent(code);
+    std::string encodedTitle = encodeURIComponent(title);
+
+    std::ostringstream urlBuilder;
+    urlBuilder << baseUrl << "/updateApptTitle?"
+                << "apptCode=" << encodedCode
+                << "&apptTitle=" << encodedTitle;
+
+    std::cout << urlBuilder.str() << std::endl;
+    RestClient::Response response = RestClient::patch(urlBuilder.str(), "application/json", "");
+
+    if (response.code == 200) {
+        auto it = std::find(DoctorClient::createdAppointments.begin(), DoctorClient::createdAppointments.end(), code);
+        if (it == DoctorClient::createdAppointments.end()) {
```

Comment:
I am not sure if i understand this part, why do we want to add the appt code if it is not found?

Reply:
Why did the client crash? I saw the service crashing. The client didn't crash for me. However I couldn't see APP1-APPT3 on client. Those are the ones read from the DB

Reply:
Just realized one thing, we probably need to retest these after DB change is in. After that, all appts will be in DB, and we can query appts based on different fields. We can then delete in memory createdAppointments